src/v/kafka/server/server.h

```diff
@@ -212,7 +212,7 @@ class server final : public net::server {
 
     handler_probe_manager _handler_probes;
```

Comment:
May of missed this one. Though all the `this` objects metric callbacks reference here are stored in a vector so they shouldn't be invalidated on moves.

Reply:
Well thankfully the vector is never resized too so the vector itself can't move them. Good catch thank you! Fixed.